### PR TITLE
Add support for watchman 4 

### PR DESCRIPTION
**Pull request by jcope2013**
update semver checks to allow for watchman 4

### CODE REVIEW

**Comment by rwjblue** [comment]
@stefanpenner - r?

**Comment by stefanpenner** [comment]
I haven't had a chance to manually confirm v4 works, I can test later today (unless @rwjblue you have a chance first, if so :shipit:

**Comment by rwjblue** [comment]
Per https://facebook.github.io/watchman/docs/compatibility.html:

> watchman does not follow semantic versioning!
>
> * Since its public APIs never make incompatible changes, MAJOR versions are moot.
> * While in the past we've released versions with three components (x.y.z), starting version 3.1 the version number will only have two components that are meaningful (x.y), with the third component always zero.
> * The version after 3.9 is expected to be 4.0. The version number string reported by these versions will be 3.9.0 and 4.0.0 respectively.

There are no breaking changes in 4.0 (they don't follow SemVer), we should update the various checks to simply do `>=`.

File: lib/models/watcher.js [hunk 1]
```diff
@@ -105,17 +105,17 @@ Watcher.detectWatcher = function(ui, _options) {
       debug('detected watchman: %s', version);
 
       var semver = require('semver');
-      if (semver.satisfies(version, '^3.0.0')) {
-        debug('watchman %s does satisfy: %s', version, '^3.0.0');
+      if (semver.satisfies(version, '>= 3.0.0 < 5.0.0')) {
```
Comment (rwjblue) [inline lib/models/watcher.js]: Per https://github.com/ember-cli/ember-cli/pull/5026#issuecomment-153366455, please change this to `>= 3.0.0`.

File: lib/models/watcher.js [hunk 2]
```diff
@@ -105,17 +105,17 @@ Watcher.detectWatcher = function(ui, _options) {
...
+        debug('watchman %s does satisfy: %s', version, '>= 3.0.0 < 5.0.0');
```
Comment (rwjblue) [inline lib/models/watcher.js]: Per https://github.com/ember-cli/ember-cli/pull/5026#issuecomment-153366455, please change this to `>= 3.0.0`.

File: lib/models/watcher.js [hunk 3]
```diff
@@ -105,17 +105,17 @@ Watcher.detectWatcher = function(ui, _options) {
...
         options._watchmanInfo = {
           enabled: true,
           version: version,
-          canNestRoots: semver.satisfies(version, '^3.7.0')
+          canNestRoots: semver.satisfies(version, '>= 3.7.0')
         };
       } else {
-        debug('watchman %s does NOT satisfy: %s', version, '^3.0.0');
-        ui.writeLine('Invalid watchman found, version: [' + version + '] did not satisfy [^3.0.0], falling back to NodeWatcher.');
+        debug('watchman %s does NOT satisfy: %s', version, '>= 3.0.0 < 5.0.0');
```
Comment (rwjblue) [inline lib/models/watcher.js]: Per https://github.com/ember-cli/ember-cli/pull/5026#issuecomment-153366455, please change this to `>= 3.0.0`.

File: lib/models/watcher.js [hunk 4]
```diff
@@ -105,17 +105,17 @@ Watcher.detectWatcher = function(ui, _options) {
...
+        ui.writeLine('Invalid watchman found, version: [' + version + '] did not satisfy [>= 3.0.0 < 5.0.0], falling back to NodeWatcher.');
```
Comment (rwjblue) [inline lib/models/watcher.js]: Per https://github.com/ember-cli/ember-cli/pull/5026#issuecomment-153366455, please change this to `>= 3.0.0`.

**Comment by jcope2013** [comment]
should be ready

**Comment by stefanpenner** [comment]
@homu r+

**Comment by rwjblue** [comment]
Thanks @jcope2013 !

**Comment by stefanpenner** [comment]
@homu retry

**Comment by benoittgt** [comment]
Thanks for this !

**Comment by alanhogan** [comment]
Might be a good practice for any `brew`-managed packages, as `brew` does not allow installation of arbitrary versions.

**Pull request merged**